File: programmers/lv1/[1차]비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    
    for i in range(n):
        binary1 = []
        binary2 = []
        row = []
        
        # arr1의 이진수 구하기
        while(arr1[i]):
            binary1.append(arr1[i] % 2)
            arr1[i] = arr1[i] // 2
        
        if len(binary1) < n: # 길이 맞추기
            while(len(binary1) < n):
                binary1.append(0)
        binary1.reverse()
        
        # arr2의 이진수 구하기
        while(arr2[i]):
            binary2.append(arr2[i] % 2)
            arr2[i] = arr2[i] // 2
            
        if len(binary2) < n: # 길이 맞추기
            while(len(binary2) < n):
                binary2.append(0)
        binary2.reverse()
        
        # 지도 합치기
        for j in range(n):
            if binary1[j] == 1 or binary2[j] == 1:
                row.append('#')
            elif binary1[j] == 0 and binary2[j] == 0:
                row.append(' ')
                
        answer.append(''.join(row))
    return answer

File: programmers/lv1/[1차]비밀지도.js
function bin(num, n) {
  let list = [];
  while (num) {
    list.push(num % 2);
    num = parseInt(num / 2);
  }
  while (list.length < n) {
    list.push(0);
  }
  return list.reverse();
}

function solution(n, arr1, arr2) {
  var answer = [];
  let graph1 = [],
    graph2 = [];

  for (let i = 0; i < arr1.length; i++) {
    const b1 = bin(arr1[i], n);
    const b2 = bin(arr2[i], n);
    graph1.push(b1);
    graph2.push(b2);
  }

  for (let i = 0; i < n; i++) {
    row = "";
    for (let j = 0; j < n; j++) {
      if (graph1[i][j] == 1 || graph2[i][j] == 1) {
        row += "#";
      } else if (graph1[i][j] == 0 || graph2[i][j] == 0) {
        row += " ";
      }
    }
    answer.push(row);
  }

  return answer;
}
